pkg/frontend/nvme_controller.go

```diff
@@ -129,11 +138,27 @@ func (s *Server) CreateNvmeController(_ context.Context, in *pb.CreateNvmeContro
 		msg := fmt.Sprintf("Could not create CTRL: %s", in.NvmeController.Name)
 		return nil, status.Errorf(codes.InvalidArgument, msg)
 	}
+
+	// TODO: this is hard coded to specific host
+	params2 := spdk.NvmfSubsystemAddHostParams{
```

Comment:
this is needed since we switched to `AllowAnyHost: false` above

Reply:
I think we could extend opiproject/opi-api#375 and wrap `FabricsEndpoint` into a `FabricsPath` containing also host nqn and psk
```proto
oneof path {
  PciPath pcie_id = 3 [(google.api.field_behavior) = OPTIONAL];
  FabricsPath fabrics_id = 4 [(google.api.field_behavior) = OPTIONAL];
}
...
message FabricsPath {
  FabricsEndpoint endpoint = 1 [(google.api.field_behavior) = REQUIRED];
  string host_nqn = 2 [(google.api.field_behavior) = REQUIRED]; // we can make it OPTIONAL later if needed. It won't be a breaking change
  bytes psk = 3 [(google.api.field_behavior) = OPTIONAL];
}

message PciePath {
  PcieEndpoint endpoint = 1 [(google.api.field_behavior) = REQUIRED];
}
```
Then we need to rework `nvmeTCPTransport` in either way:
1. It returns a list of cmds with parameters to execute for Create and Delete
``` go
type spdkCmd struct {
  callName string
  params any
}
...
func (c *nvmeTCPTransport) CreateParams(ctrlr *pb.NvmeController, nqn string) ([]spdkCmd, error) {
  result := []spdkCmd{}
  // add nvmf_subsystem_add_listener cmd to result
  // add nvmf_subsystem_add_host cmd to result
  if len(ctrlr. ... .Psk) != 0 {
    // add secure channel to nvmf_subsystem_add_listener
    // set psk to nvmf_subsystem_add_host 
  }
  return result, nil
}
func (c *nvmeTCPTransport) DeleteParams(ctrlr *pb.NvmeController, nqn string) (spdkCmd, error) {
  return /* nvmf_subsystem_remove_listener cmd*/, nil
}
```
and in `CreateNvmeController` execute returned cmds one by one

2. Another option is to move SPDK calls into NvmeTransports instead of returning parameters
```go
type NvmeTransport interface {
    // Params(ctrlr *pb.NvmeController, nqn string) (spdk.NvmfSubsystemAddListenerParams, error)
	CreateController(ctrlr *pb.NvmeController, nqn string) error
    DeleteController(ctrlr *pb.NvmeController, nqn string) error
}
```